consensus/scp/tests/test_cyclic_networks.rs

```diff
@@ -25,76 +13,52 @@ fn skip_slow_tests() -> bool {
 // Cyclic tests (similar to Figure 4 in the SCP whitepaper)
 ///////////////////////////////////////////////////////////////////////////////
 
-fn cyclic_test_helper(num_nodes: usize, logger: Logger) {
-    if skip_slow_tests() {
-        return;
+/// Constructs a cyclic network (e.g. 1->2->3->4->1)
+fn new_cyclic(
+    num_nodes: usize,
+    test_options: mock_network::TestOptions,
+    logger: Logger,
+) -> mock_network::SCPNetwork {
+    let mut node_options = Vec::<mock_network::NodeOptions>::new();
+    for node_id in 0..num_nodes {
+        let next_node_id: u32 = if node_id + 1 < num_nodes {
+            node_id as u32 + 1
+        } else {
+            0
+        };
+
+        let other_node_ids: Vec<u32> = (0..num_nodes)
+            .filter(|other_node_id| other_node_id != &node_id)
+            .map(|other_node_id| other_node_id as u32)
+            .collect();
+
+        node_options.push(mock_network::NodeOptions::new(
+            format!("c-{}-node{}", num_nodes, node_id),
+            other_node_ids,
+            vec![next_node_id],
+            1,
+        ));
     }
 
-    assert!(num_nodes >= 3);
-    let mut rng: StdRng = SeedableRng::from_seed([193u8; 32]);
-    let start = Instant::now();
-
-    let network = mock_network::SCPNetwork::new_cyclic(
-        num_nodes,
-        Arc::new(test_utils::trivial_validity_fn::<String>),
-        //                Arc::new(test_utils::get_bounded_combine_fn::<String>(200)),
-        Arc::new(test_utils::trivial_combine_fn::<String>),
-        logger.clone(),
-    );
-
-    // Send a few values, with random timeouts in between
-    let mut values = Vec::<String>::new();
+    mock_network::SCPNetwork::new(node_options, test_options, logger)
+}
 
-    for _i in 0..10 {
-        let n = test_utils::test_node_id(rng.gen_range(0, num_nodes as u32));
-        for _j in 0..1000 {
-            let value = format!("{}-{}", n, mock_network::random_str(&mut rng, 10));
-            network.push_value(&n, &value);
-            values.push(value);
-        }
-        sleep(Duration::from_millis(rng.gen_range(0, 50)));
+/// Performs a consensus test for a cyclic network of `num_nodes` nodes.
+fn cyclic_test_helper(num_nodes: usize, logger: Logger) {
+    if num_nodes < 3 {
+        return;
```

Comment:
This should be an assert.

Reply:
returning allows this function to be called in a loop over network size without worrying about num_nodes... but I'm okay with panic if you prefer.

Reply:
I don't feel strongly

Reply:
let's assert; people shouldn't loop over tests that don't run